### PR TITLE
Compression refactoring

### PR DESCRIPTION
**Pull request by Tinsane**
Refactored compression logic to a separate folder.
Removed ReaderFromWriteCloser to adhere DRY.
Removed CompressingPipeWriter antipattern.

### CODE REVIEW

**Comment by x4m** [comment]
Hi! While refactoring looks good and useful, CI is unhappy.

File: internal/compress_and_encrypt.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	return CompressAndEncryptError{errors.New(reason)}
+}
+
+func (err CompressAndEncryptError) Error() string {
+	return fmt.Sprintf(tracelog.GetErrorFormatter(), err.error)
+}
+
+// CompressAndEncrypt compresses input to a pipe reader. Output must be used or
+// pipe will block.
+func CompressAndEncrypt(source io.Reader, compressor compression.Compressor, crypter Crypter) io.Reader {
```
Comment (x4m) [inline internal/compress_and_encrypt.go]: Eventually we will have to stop passing crypter here and there...

File: testtools/util.go [hunk 2]
```diff
@@ -75,6 +66,14 @@ func AssertReaderIsEmpty(t *testing.T, reader io.Reader) {
 	assert.Equal(t, io.EOF, err)
 }
 
+type NopCloserWriter struct {
```
Comment (x4m) [inline testtools/util.go]: Don't we have this in ioutils?

Comment (Tinsane) [reply testtools/util.go]: Nop

**Pull request merged**